Sources/NIOCore/ChannelHandlers.swift

```diff
@@ -294,42 +304,45 @@ public final class IdleStateHandler: ChannelDuplexHandler, RemovableChannelHandl
         }
     }
 
-    private func makeAllTimeoutTask(_ context: ChannelHandlerContext, _ timeout: TimeAmount) -> (() -> Void) {
+    private func makeAllTimeoutTask(_ context: ChannelHandlerContext, _ timeout: TimeAmount) -> TimeoutTask {
         return {
             guard self.shouldReschedule(context) else  {
                 return
             }
 
             if self.reading {
-                self.scheduledReaderTask = context.eventLoop.scheduleTask(in: timeout, self.makeAllTimeoutTask(context, timeout))
+                self.scheduledAllTask = context.eventLoop.scheduleTask(in: timeout, self.makeAllTimeoutTask(context, timeout))
                 return
             }
             let lastRead = self.lastReadTime
             let lastWrite = self.lastWriteCompleteTime
             let latestLast = max(lastRead, lastWrite)
 
-            let diff = .now() - latestLast
+            let diff = self.now() - latestLast
             if diff >= timeout {
-                // Reader is idle - set a new timeout and trigger an event through the pipeline
-                self.scheduledReaderTask = context.eventLoop.scheduleTask(in: timeout, self.makeAllTimeoutTask(context, timeout))
+                // Both reader and writer are idle - set a new timeout and trigger an event through the pipeline
+                self.scheduledAllTask = context.eventLoop.scheduleTask(in: timeout, self.makeAllTimeoutTask(context, timeout))
 
                 context.fireUserInboundEventTriggered(IdleStateEvent.all)
             } else {
-                // Read occurred before the timeout - set a new timeout with shorter delay.
-                self.scheduledReaderTask = context.eventLoop.scheduleTask(deadline: latestLast + timeout, self.makeAllTimeoutTask(context, timeout))
+                // Read or write occurred before the timeout - set a new timeout with shorter delay.
+                self.scheduledAllTask = context.eventLoop.scheduleTask(deadline: latestLast + timeout, self.makeAllTimeoutTask(context, timeout))
             }
         }
     }
 
-    private func schedule(_ context: ChannelHandlerContext, _ amount: TimeAmount?, _ body: @escaping (ChannelHandlerContext, TimeAmount) -> (() -> Void) ) -> Scheduled<Void>? {
+    private typealias TimeoutTaskFactory = (ChannelHandlerContext, TimeAmount) -> TimeoutTask
```

Comment:
Not sure what a good name for this could be, `Factory` sounds a little Javaesque, @Lukasa let me know if you have a better name in mind, or if I should revert these type aliases back. 😅